Providers/PocketFeed.swift

```diff
@@ -0,0 +1,134 @@
+private let MaxCacheAge = OneMinuteInMilliseconds * 60 // 1 hour in seconds
+
+/*s
+ The Pocket class is used to fetch stories from the Pocked API.
+ Right now this only supports the global feed
+
+ For a sample feed item check ClientTests/pocketglobalfeed.json
+ */
+struct PocketStory {
+    let url: URL
+    let title: String
+    let storyDescription: String
+    let imageURL: URL
+    let domain: String
+
+    static func parseJSON(list: Array<[String: Any]>) -> [PocketStory] {
+        return list.flatMap({ (storyDict) -> PocketStory? in
+            guard let urlS = storyDict["url"] as? String, let domain = storyDict["domain"] as? String,
+                let imageURLS = storyDict["image_src"] as? String,
+                let title = storyDict["title"] as? String,
+                let description = storyDict["excerpt"] as? String else {
+                    return nil
+            }
+            guard let url = URL(string: urlS), let imageURL = URL(string: imageURLS) else {
+                return nil
+            }
+            return PocketStory(url: url, title: title, storyDescription: description, imageURL: imageURL, domain: domain)
+        })
+    }
+}
+
+private class PocketError: MaybeErrorType {
+    var description = "Failed to load from API"
+}
+
+class Pocket {
+
+    let pocketGlobalFeed: String
+    // Allow endPoint to be overriden for testing
+    init(endPoint: String? = nil) {
+        pocketGlobalFeed = endPoint ?? PocketGlobalFeed
+    }
+
+    lazy fileprivate var alamofire: SessionManager = {
+        let ua = UserAgent.fxaUserAgent //TODO: use a different UA
+        let configuration = URLSessionConfiguration.default
+        return SessionManager.managerWithUserAgent(ua, configuration: configuration)
+    }()
+
+    func findCachedResposneFor(request: URLRequest) -> [String: Any]? {
+        let cachedResponse = URLCache.shared.cachedResponse(for: request)
+        guard let cachedAtTime = cachedResponse?.userInfo?["cache-time"] as? Timestamp, (Date.now() - cachedAtTime) < MaxCacheAge else  {
+            return nil
+        }
+
+        guard let data = cachedResponse?.data, let json = try? JSONSerialization.jsonObject(with: data, options: []) else {
+            return nil
+        }
+
+        return json as? [String: Any]
+    }
+
+    func cacheResponseFor(response: HTTPURLResponse?, request: URLRequest, data: Data?) {
+        guard let resp = response, let data  = data else {
+            return
+        }
+        let metadata = ["cache-time": Date.now()]
+        let cachedResp = CachedURLResponse(response: resp, data: data, userInfo: metadata, storagePolicy: .allowed)
+        URLCache.shared.removeCachedResponse(for: request)
+        URLCache.shared.storeCachedResponse(cachedResp, for: request)
+    }
+
+    // Fetch items from the global pocket feed
+    func globalFeed(items: Int = 2) -> Deferred<Maybe<Array<PocketStory>>> {
+        let deferred = Deferred<Maybe<Array<PocketStory>>>()
```

Comment:
👍  I would do the same to follow an existing style --and is orthogonal topic to the work on this bug